app/src/main/res/layout/tool_details_fragment.xml

```diff
@@ -56,95 +62,101 @@
+                    android:id="@+id/action_open"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginLeft="@dimen/tool_details_margin_horizontal"
+                    android:layout_marginTop="16dp"
+                    android:layout_marginRight="@dimen/tool_details_margin_horizontal"
+                    android:onClick="@{() -> fragment.openTool(tool, primaryTranslation, parallelTranslation)}"
+                    android:text="@string/action_tools_open"
+                    app:layout_constraintTop_toBottomOf="@id/shares"
+                    app:layout_goneMarginTop="16dp"
+                    app:visibleIf="@{tool.isAdded}" />
+
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/action_add"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginLeft="@dimen/tool_details_margin_horizontal"
+                    android:layout_marginRight="@dimen/tool_details_margin_horizontal"
+                    android:onClick="@{() -> fragment.addTool(tool.code)}"
+                    android:text="@string/action_tools_download"
+                    android:theme="@style/ThemeOverlay.GodTools.Button.Green"
+                    app:layout_constraintTop_toBottomOf="@id/action_open"
+                    app:visibleIf="@{!tool.isAdded}" />
+
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/action_remove"
+                    style="?attr/materialButtonOutlinedStyle"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginLeft="@dimen/tool_details_margin_horizontal"
+                    android:layout_marginRight="@dimen/tool_details_margin_horizontal"
+                    android:onClick="@{() -> fragment.removeTool(tool.code)}"
+                    android:text="@string/action_tools_remove"
+                    android:theme="@style/ThemeOverlay.GodTools.Button.Red"
+                    app:layout_constraintTop_toBottomOf="@id/action_add"
+                    app:visibleIf="@{tool.isAdded()}" />
+
+                <com.google.android.material.tabs.TabLayout
+                    android:id="@+id/tabs"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="32dp"
+                    app:layout_constraintTop_toBottomOf="@id/action_remove"
+                    app:tabGravity="center" />
+            </androidx.constraintlayout.widget.ConstraintLayout>
 
-            <org.ccci.gto.android.common.viewpager.view.ChildHeightAwareViewPager
-                android:id="@+id/detail_view_pager"
+            <androidx.viewpager2.widget.ViewPager2
```

Comment:
I purposely left this out of the `ConstraintLayout` so that we could add elevation to the `ConstraintLayout` for the shadow between the `TabLayout` and `ViewPager2`